plugins/Demo/DateSetting/DateSetting.php

```diff
@@ -0,0 +1,114 @@
+<?php
+
+/**
+ * DateSetting Plugin for LimeSurvey
+ * @author : Denis Chenu
+ * @copyright: LimeSurvey <https://community.limesurvey.org/>
+ * @version:; 0.1.0
+ * @licence MIT Licence
+ */
+class TestDateSetting extends PluginBase
+{
+    /** @inheritdoc **/
+    protected static $name = 'TestDateSetting';
+    /** @inheritdoc **/
+    protected static $description = 'Pluginto test HTML DB setting.';
+    /** @inheritdoc **/
+    protected $storage = 'DbStorage';
+    /** @inheritdoc **/
+    protected $settings = [
+        'checkDateDate' => array(
+            'type' => 'date',
+            'label' => 'A date only saved',
+            'saveformat' => "Y-m-d",
+        ),
+        'checkDateYear' => array(
+            'type' => 'date',
+            'label' => 'A year only saved',
+            'saveformat' => "Y",
+        ),
+        'checkDateDefault' => array(
+            'type' => 'date',
+            'label' => 'Another date no format set',
+        ),
+        'checkDateFalse' => array(
+            'type' => 'date',
+            'label' => 'A date saved as shown',
```

Comment:
Not needed, but I think we could show on helptext the value stored on the DB, as to show the difference in between how is selected and how is saved.

Reply:
You mean when reload ? Or in help text ?

Reply:
Added in help text (and with translation for SurveySettings)

Related issue : https://bugs.limesurvey.org/view.php?id=18668

Reply:
> You mean when reload ? Or in help text ?

I was thinkin in help text having something like `"last saved value on DB is " . $this->get('checkSurveyDateDefault', 'Survey', $surveyId, '') `. 

So admins can see how the value is actually being saved on the db

Reply:
Done : https://github.com/LimeSurvey/LimeSurvey/pull/2753/commits/1bd693fcedac15cd240bb4341c00f46c16b9ff43